src/game/GameResults.as

```diff
@@ -1150,6 +1153,33 @@ package game
                             newLevelRanks["fcs"] += previousLevelRanks["fcs"];
                         }
                         _gvars.activeUser.level_ranks[songInfo.level] = newLevelRanks;
+
+                        // Update/replace Skill Rating top X song if better than our lowest SR equiv
+                        if (newLevelRanks.equiv > _gvars.activeUser.skill_rating_songs[_gvars.activeUser.skill_rating_top_count - 1].equiv)
```

Comment:
Suggestion: I'd move this whole logic to a function on the User type. Not only will it make it much less verbose without the may `_gvars.activeuser...` but you'll keep this GameResults logic more concise.

Reply:
Agreed, have made this modification in the below commit